File: 05-Proyecto/Backend/canchas/api/models/Cliente.js
/**
 * Cliente.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {
  
    cedula: {
      type: 'string',
      required: true,
      unique: true,
      minLength: 10,
      maxLength: 25,
    },
    nombre: {
      type: 'string',
      required: true,
      minLength: 3,
      maxLength: 15,
    },
    telefono: {
      type: 'string',
      required: false,
      minLength: 3,
      maxLength: 15,
    },
    direccion: {
      type: 'string',
      required: true,
      minLength: 3,
      maxLength: 15,
    },
    compraDeCliente: {     // Nombre atributo de la relación
      collection: 'reserva', // Nombre del modelo a relacionar
      via: 'codigoCli'        // Nombre del campo a hacer la relacion
    },
  },
};

File: 05-Proyecto/ProyectoIndividual/app/src/main/java/com/example/proyectoindividual/modulo_clientes/Cliente.kt
package com.example.proyectoindividual.modulo_clientes

class Cliente(
//    var compraDeCliente: ArrayList<CompraAux>?,
    var createdAt: Long?,
    var updatedAt: Long?,
    var id: Int?,
    var nombre: String,
    var cedula: String,
    var telefono: String,
    var direccion: String
) {

}


//parcel.readSerializable() as? ArrayList<Reserva>,
//parcel.writeSerializable(compraDeCliente)